src/compiler/checker.ts

```diff
@@ -25066,9 +25066,13 @@ namespace ts {
             }
 
             const localOrExportSymbol = getExportSymbolOfValueSymbolIfExported(symbol);
-            const sourceSymbol = localOrExportSymbol.flags & SymbolFlags.Alias ? resolveAlias(localOrExportSymbol) : localOrExportSymbol;
-            if (sourceSymbol.declarations && getDeclarationNodeFlagsFromSymbol(sourceSymbol) & NodeFlags.Deprecated && isUncalledFunctionReference(node, sourceSymbol)) {
-                addDeprecatedSuggestion(node, sourceSymbol.declarations, node.escapedText as string);
+            if (localOrExportSymbol.flags & SymbolFlags.Alias) {
+                checkDeprecatedAliasedSymbol(localOrExportSymbol, node);
+            }
+            else {
+                if (localOrExportSymbol.declarations && getDeclarationNodeFlagsFromSymbol(localOrExportSymbol) & NodeFlags.Deprecated && isUncalledFunctionReference(node, localOrExportSymbol)) {

```

Comment:
this line should be broken up

Reply:
Moved to the new helper `isDeprecatedSymbol`